src/main/java/com/github/bartimaeusnek/cropspp/crops/TF/KnighmetalCrop.java

```diff
@@ -42,14 +37,20 @@ protected String hasBlock() {
     @Override
     public int growthDuration(ICropTile crop) {
         if (ConfigValues.debug) return 1;
-        return crop.getSize() >= 2 ? 4500 : 1000;
+        return crop.getSize() >= this.maxSize() - 1 ? 4500 : 1000;
     }
 
     @Override
     public ItemStack getGain(ICropTile crop) {
-        if (crop.getSize() == 4
-                && (crop.isBlockBelow("blockKnightmetal") || !OreDictionary.doesOreNameExist("blockKnightmetal"))) {
-            return new ItemStack(twilightforest.item.TFItems.armorShard).splitStack(4);
-        } else return new ItemStack(twilightforest.item.TFItems.armorShard);
+        if (crop.getSize() < this.maxSize()
+                || !((crop.isBlockBelow(this.hasBlock()) || !OreDictionary.doesOreNameExist(this.hasBlock())))) {
```

Comment:
This is less readable imo, double negation

Reply:
should have been addressed by https://github.com/GTNewHorizons/Crops-plus-plus/pull/57/commits/38c9838a209ddeed10cd00bfea8786ac8a2a3baf